cypress/e2e/tests/pages/generic/login.spec.ts

```diff
@@ -36,16 +43,31 @@ describe('Local authentication', { tags: ['@generic', '@adminUser', '@standardUs
 
   it('Log in with valid credentials', () => {
     LoginPagePo.goTo();
+
     cy.intercept('POST', '/v3-public/localProviders/local*').as('loginReq');
 
+    const loginPage = new LoginPagePo();
+
+    // take screenshot if locator is visible.
+    loginPage.canSubmit().then((canSubmit) => {
+      if (canSubmit) {
+        // Take a snapshot for visual diffing
+        cy.percySnapshot('Login test');
```

Comment:
Using an external service like this could trivialize the screenshots storage for diff, and the reporting looks great too.  Also it looks really straightforward to take the snapshots.